### ISSUE TITLE
Error messages for bitfield attributes always use the span of the top-level bitfield macro

### ISSUE THREAD

**Issue by vortexofdoom** [danlehmann/bitfield#31]
For a struct like:
```
#[bitfield(u8)]
struct Foo {
    /*...*/
    #[bits(0..3, rw)]
    bar: u2,
   /*...*/
}
```
The error given will show up as:
```
error: custom attribute panicked
  --> src/foo.rs:x:1
   |
 x | #[bitfield(u8)]
   | ^^^^^^^^^^^^^^^
   |
   = help: message: bitfield!: Expected inclusive range, e.g. bits(1..=8, rw)'
```
rather than:
```
error: custom attribute panicked
  --> src/foo.rs:y:1
   |
 y | #[bits(0..3, rw)]
   |        ^^^^
   |
   = help: message: bitfield!: Expected inclusive range, e.g. bits(1..=8, rw)'
```

Even if the span were from the field attribute instead of the struct attribute it'd be a little more intuitive.

I think this would probably involve manually constructing a compilation error instead of just panicking? I'm willing to take a crack at it but figured I'd start by documenting.

**Comment by danlehmann** [comment]
I agree this is annoying! It had never occurred to me an error message could be constructed manually. I assume it would have to be written in a style that looks like the regular panic. How would you know the real line number though?

**Comment by vortexofdoom** [comment]
`syn::Error::new_spanned` takes in something spanned and outputs an error corresponding to it.

`RangeExpr` would very possibly be the best way to make it point directly at that segment as I suggested, but just passing in the `Attribute` itself will at the least highlight the correct area.

**Comment by vortexofdoom** [comment]
It looks to me like the `Attribute` path would be the easiest way as it's currently laid out, since all processing past that point is direct string manipulation.

But it could technically use `syn::parse` to get a more generic parse that just gave back each element with a little more semantic meaning (that didn't require special-casing "bit" vs "bits" when parsing the rest for instance). That would require something of a refactor though.
